source/tutorials/learning-journey/sharing-dependencies.md

```diff
@@ -0,0 +1,204 @@
+
+However, now you need to define a `shell.nix` that not only provides your development packages, but can also build your project.
+In other words, you need a `shell.nix` that brings in all of the packages that your build depends on.
+You could certainly copy the build dependencies from `default.nix` and copy them into `shell.nix`, but this is less than ideal:
+your build dependencies are defined in multiple places, and aside from repeating yourself there's now the possiblity that the dependencies in `default.nix` and `shell.nix` may fall out of sync.
+
+There is a better way!
+
+## Getting started
+
+Create a directory called `shared_project` and enter it:
+
+```console
+$ mkdir shared_project
+$ cd shared_project
+```
+
+You'll be creating a Python web application as an example project, but don't worry, you'll be given all of the code you need and won't need to know Python to proceed.
+
+Create a new directory called `src` and two empty files inside of `src` called `__init__.py` and `app.py`:
+
+```
+$ mkdir src
+$ touch src/__init__.py
+$ touch src/app.py
+```
+
+Copy the following contents into `app.py`:
+
+```python
+from flask import Flask
+
+app = Flask(__name__)
+
+@app.route("/")
+def hello_world():
+    return "<p>Hello, World!</p>"
+```
+
+This creates a web application that returns `<p>Hello, World!</p>` on the `/` route.
+
+Next create a `pyproject.toml` file with the following contents:
+
+```toml
+[build-system]
+requires = ["setuptools", "setuptools-scm"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "shared_project"
+version = "0.0.1"
+
+[project.scripts]
+app = "app:main"
+```
+
+This file tells Python how to build the project and what will execute when you run the executable called `app`.
+
+For the Nix part of the project you'll create two files: `build.nix` and `default.nix`.
+The actual build recipe will be in `build.nix` and `default.nix` will import this file to perform the build.
+
+First create a `build.nix` file like this:
+
+```nix
+{
+  buildPythonApplication,
+  setuptools-scm,
+  flask,
+}:
+
+buildPythonApplication {
+  pname = "shared_project";
+  version = "0.0.1";
+  format = "pyproject";
+  src = builtins.path { path = ./.; name = "shared_project_source"; };
+  propagatedBuildInputs = [
+    setuptools-scm
+    flask
+  ];
+}
+```
+
+The Nix expression in this file is a _function_ that produces a derivation.
+This method of defining builds is a common design pattern in the Nix community, and is the format used throughout the `nixpkgs` repository.
+This particular derivation builds your Python application and ensures that `flask`, the library used to create the web application, is available at runtime for the application.
+
+Note that on line 11 of the `build.nix` file the `src` attribute is set using `builtins.path`.
+This is a good habit to form because it will give your build a fixed name rather than simply inheriting the name of the parent directory.
```

Comment:
```suggestion
Note that on line 11 of the `build.nix` file the `src` attribute is set using `builtins.path` to ensure that the store path of the build's [source directory stays independent of the containing directory name](...).
```

We can just link to that page here: https://nix.dev/recipes/best-practices#reproducible-source-paths